repositories.bzl

```diff
@@ -0,0 +1,559 @@
+        "src/c++11",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@lightstep_common_git//:collector_proto",
+        ":envoy_carrier_proto",
+        "//external:protobuf",
+    ],
+)"""
+
+    COMMON_BUILD = """
+load("@protobuf_git//:protobuf.bzl", "cc_proto_library")
+
+genrule(
+    name = "collector_pb",
+    srcs = ["collector.proto"],
+    outs = ["lightstep/collector.proto"],
+    cmd = "cp $(SRCS) $@",
+)
+
+cc_proto_library(
+    name = "collector_proto",
+    srcs = ["lightstep/collector.proto"],
+    include = ".",
+    deps = [
+        "//external:cc_wkt_protos",
+    ],
+    protoc = "//external:protoc",
+    default_runtime = "//external:protobuf",
+    visibility = ["//visibility:public"],
+)"""
+
+    native.new_git_repository(
+        name = "lightstep_common_git",
+        remote = "https://github.com/lightstep/lightstep-tracer-common.git",
+        commit = "8d932f7f76cd286691e6179621d0012b0ff1e6aa",
+        build_file_content = COMMON_BUILD,
+    )
+
+    native.new_git_repository(
+        name = "lightstep_git",
+        remote = "https://github.com/lightstep/lightstep-tracer-cpp.git",
+        commit = "5a71d623cac17a059041b04fabca4ed86ffff7cc",
+        build_file_content = BUILD,
+    )
+
+    if bind:
+        native.bind(
+            name = "lightstep",
+            actual = "@lightstep_git//:lightstep_core",
+        )
+
+def http_parser_repositories(bind=True):
+    BUILD = """
+cc_library(
+    name = "http_parser",
+    srcs = [
+        "http_parser.c",
+    ],
+    hdrs = [
+        "http_parser.h",
+    ],
+    visibility = ["//visibility:public"],
+)"""
+
+    native.new_git_repository(
+        name = "http_parser_git",
+        remote = "https://github.com/nodejs/http-parser.git",
+        commit = "9b0d5b33ebdaacff1dadd06bad4e198b11ff880e",
+        build_file_content = BUILD,
+    )
+
+    if bind:
+        native.bind(
+            name = "http_parser",
+            actual = "@http_parser_git//:http_parser",
+        )
+
+def rapidjson_repositories(bind=True):
+    BUILD = """
+cc_library(
+    name = "rapidjson",
+    srcs = glob([
+        "include/rapidjson/internal/*.h",
+    ]),
+    hdrs = glob([
+        "include/rapidjson/*.h",
+        "include/rapidjson/error/*.h",
+    ]),
+    includes = ["include"],
+    visibility = ["//visibility:public"],
+)
+"""
+
+    native.new_git_repository(
+        name = "rapidjson_git",
+        remote = "https://github.com/miloyip/rapidjson.git",
+        commit = "f54b0e47a08782a6131cc3d60f94d038fa6e0a51", # v1.1.0
+        build_file_content = BUILD,
+    )
+
+    if bind:
+        native.bind(
+            name = "rapidjson",
+            actual = "@rapidjson_git//:rapidjson",
+        )
+
+def nghttp2_repositories(bind=True):
+    BUILD = """
+genrule(
+    name = "config",
+    srcs = glob([
+        "**/*",
+    ]),
+    outs = [
+        "config.h",
+    ],
+    tools = [
+        "configure",
+    ],
+    cmd = ' '.join([
```

Comment:
Buildifier: `s/' '/" "/g`.

Also, I personally prefer concatenating strings, instead of joining them:

    cmd = "$(location configure) --enable-lib-only --enable-shared=no" +
          " && cp config.h $@",

instead of:

    cmd = ' '.join([
          "$(location configure) --enable-lib-only --enable-shared=no",
          "&& cp config.h $@",
    ]),